Examples/ReloadDataExample/ReloadDataViewController.swift

```diff
@@ -0,0 +1,151 @@
+    return button
+  }()
+
+  let removeButton: UIButton = {
+    let button = UIButton()
+    let removeAttributedTitle =
+      NSAttributedString(string: "-",
+                         attributes: [NSFontAttributeName : UIFont.boldSystemFont(ofSize: 20)])
+    button.setAttributedTitle(removeAttributedTitle, for: .normal)
+    button.setTitleColor(.black, for: .normal)
+    button.backgroundColor = UIColor(hue: 0.20, saturation: 0.68, brightness: 0.98, alpha: 1)
+    return button
+  }()
+
+  let reloadButton: UIButton = {
+    let button = UIButton()
+    button.setTitle("Reset", for: .normal)
+    button.setTitleColor(.black, for: .normal)
+    button.backgroundColor = UIColor(hue: 0.22, saturation: 0.68, brightness: 0.98, alpha: 1)
+    return button
+  }()
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    view.backgroundColor = .white
+    addButton.addTarget(self, action: #selector(add), for: .touchUpInside)
+    view.addSubview(addButton)
+    removeButton.addTarget(self, action: #selector(remove), for: .touchUpInside)
+    view.addSubview(removeButton)
+    reloadButton.addTarget(self, action: #selector(reload), for: .touchUpInside)
+    view.addSubview(reloadButton)
+
+    let dataProvider = ArrayDataProvider(data: data, identifierMapper: { (_, data) in
+      return "\(data)"
+    })
+    let layout = FlowLayout<Int>(insets: UIEdgeInsetsMake(15, 15, 15, 15),
+                                 lineSpacing: 15,
+                                 interitemSpacing: 15)
+    let presenter = ZoomPresenter()
+    presenter.insertAnimation = .scale
+    presenter.deleteAnimation = .scale
+    provider = CollectionProvider(
+      dataProvider: dataProvider,
+      viewUpdater: { (view: UILabel, data: Int, index: Int) in
+        view.backgroundColor = UIColor(hue: CGFloat(data) / CGFloat(maxDataCount),
+                                       saturation: 0.68,
+                                       brightness: 0.98,
+                                       alpha: 1)
+        view.textColor = .white
+        view.textAlignment = .center
+        view.text = "\(data)"
+      },
+      layout: layout,
+      sizeProvider: { _, _, _ in
+        return CGSize(width: 60, height: 60)
+      },
+      presenter: presenter
+    )
+  }
+
+  override func viewDidLayoutSubviews() {
+    super.viewDidLayoutSubviews()
+    let viewWidth = view.bounds.width
+    let viewHeight = view.bounds.height
+    addButton.frame = CGRect(x: 0, y: 0, width: 80, height: 44)
+    removeButton.frame = CGRect(x: viewWidth - 80, y: 0, width: 80, height: 44)
+    reloadButton.frame = CGRect(x: 80, y: 0, width: viewWidth - 80 * 2, height: 44)
+    collectionView.frame = CGRect(x: 0, y: 44, width: viewWidth, height: viewHeight - 44)
+  }
+}
+
+extension ReloadDataViewController {
+  private func updateDataProvider() {
+    if let collectionProvider = provider as? CollectionProvider<Int, UILabel> {
+      collectionProvider.dataProvider =
```

Comment:
Ideally you should store a data provider, and you can add or remove from the data provider's .data array. This way it will automatically trigger an update to the collectionView. and you don't need any casting.